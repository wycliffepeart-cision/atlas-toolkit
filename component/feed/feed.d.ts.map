{"version":3,"file":"feed.d.ts","sourceRoot":"","sources":["../../../src/component/feed/feed.ts"],"names":[],"mappings":"AAGA,qBACa,IAAK,SAAQ,WAAW;CAAG"}