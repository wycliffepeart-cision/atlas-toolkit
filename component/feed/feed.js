import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var FeedTemplate = "<template>\n  <slot name=\"prefix\" part=\"prefix\"></slot>\n  <slot name=\"filter\" part=\"filter\"></slot>\n  <slot name=\"body\" part=\"body\"></slot>\n</template>\n";

let Feed = class Feed extends HTMLElement {
};
Feed = __decorate([
    Template(FeedTemplate)
], Feed);
if (!customElements.get('atlas-feed'))
    customElements.define('atlas-feed', Feed);

export { Feed };
