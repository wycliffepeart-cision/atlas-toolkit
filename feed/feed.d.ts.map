{"version":3,"file":"feed.d.ts","sourceRoot":"","sources":["../../src/feed/feed.ts"],"names":[],"mappings":"AAGA,qBACa,IAAK,SAAQ,WAAW;CAAG"}