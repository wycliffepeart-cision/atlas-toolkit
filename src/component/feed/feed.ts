import FeedTemplate from './feed.html';
import { Template } from '../../decorators/template';

@Template(FeedTemplate)
export class Feed extends HTMLElement {}

if (!customElements.get('atlas-feed')) customElements.define('atlas-feed', Feed);