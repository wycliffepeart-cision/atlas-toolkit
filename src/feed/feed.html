<template>
  <slot name="prefix" part="prefix"></slot>
  <slot name="filter" part="filter"></slot>
  <slot name="body" part="body"></slot>
</template>
