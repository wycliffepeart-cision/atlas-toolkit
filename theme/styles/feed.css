@layer atlas {
	atlas-feed {
		--default-background-color: var(--atlas-color-primary-n50);
		--default-container-width-size: 300px;

		--background-color-container: var(--default-background-color);
		background-color: var(--background-color-container);
		padding: var(--desktop-common-block-start) 1em 1em 1em;
		display: block;
		height: 100vh;
		overflow-y: auto;
		width: var(--default-container-width-size);
		box-sizing: border-box;
	}

	atlas-feed::part(prefix) {
		display: flex;
	}
}
