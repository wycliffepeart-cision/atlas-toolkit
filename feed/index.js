import { Feed } from './feed.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-feed'))
    customElements.define('atlas-feed', Feed);
