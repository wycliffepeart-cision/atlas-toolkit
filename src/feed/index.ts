import { Feed } from './feed';

if (!customElements.get('atlas-feed')) customElements.define('atlas-feed', Feed);
