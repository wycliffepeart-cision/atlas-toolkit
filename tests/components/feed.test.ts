import '../../src/component/feed';

const ELEMENT_TAG = 'atlas-feed';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;
const shadowContentSnapshot = `
<template>
  <slot name="prefix"></slot>
  <slot name="body"></slot>
</template>
`;

describe('Feed', () => {
  it('is correctly appended in the DOM', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    const selectedElement = document.querySelector(ELEMENT_TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toContainHTML('<atlas-feed></atlas-feed>');
    (expect(element) as any).toBeEmptyDOMElement();
    (expect(getShadowDomContent(element)) as any).toMatchSnapshot(shadowContentSnapshot);
  });
});
