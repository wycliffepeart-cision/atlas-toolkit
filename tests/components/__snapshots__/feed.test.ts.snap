// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed has the correct content when created: 
<template>
  <slot name="prefix"></slot>
  <slot name="body"></slot>
</template>
 1`] = `
<slot name="prefix"
      part="prefix"
>
</slot>
<slot name="filter"
      part="filter"
>
</slot>
<slot name="body"
      part="body"
>
</slot>
`;
